exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/FunctionsIterator.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.drill.exec.store.sys.store.DataChangeVersion;
+import org.apache.drill.shaded.guava.com.google.common.collect.ListMultimap;
+
+/**
+ * List functions as a System Table
+ */
+public class FunctionsIterator implements Iterator<Object> {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(FunctionsIterator.class);
+
+  private LocalFunctionRegistry localFunctionRegistry;
+  private RemoteFunctionRegistry remoteFunctionRegistry;
+  private Iterator<FunctionInfo> sortedIterator;
+
+  public FunctionsIterator(ExecutorFragmentContext context) {
+    Map<String, FunctionInfo> functionMap = new HashMap<String, FunctionInfo>();
+
+    this.localFunctionRegistry = context.getLocalFunctionRegistry();
+    listLocalFunctionRegistry(functionMap);
+
+    this.remoteFunctionRegistry = context.getRemoteFunctionRegistry();
+    listRemoteFunctionRegistry(functionMap);
+
+    //Extract list & sort via comparator
+    List<FunctionInfo> functionList = new ArrayList<FunctionsIterator.FunctionInfo>(functionMap.values());
+    functionList.sort(initFunctionComparator());
+    sortedIterator = functionList.iterator();
+  }
+
+  //Provide a function comparator
+  private Comparator<FunctionInfo> initFunctionComparator() {
+    //Initialize the comparator
+    return new Comparator<FunctionInfo>() {
+      @Override
+      public int compare(FunctionInfo o1, FunctionInfo o2) {
+        int result = o1.name.compareTo(o2.name);
+        if (result == 0) {
+          result = o1.signature.compareTo(o2.signature);
+        }
+        if (result == 0) {
+          return o1.returnType.compareTo(o2.returnType);
+        }
+        return result;
+      }
+    };
+  }
+
+  //Loads local-registry functions into functionMap
+  private void listLocalFunctionRegistry(Map<String, FunctionInfo> functionMap) {
+    ListMultimap<String, DrillFuncHolder> builtInFuncHolderList = localFunctionRegistry.getRegistryHolder().getAllFunctionsWithHolders();
+    logger.debug("{} functions are currently initialized", builtInFuncHolderList.size());
+
+    Iterator<String> builtInFuncIter = builtInFuncHolderList.asMap().keySet().iterator();
+    while (builtInFuncIter.hasNext()) {
+      String s = builtInFuncIter.next();
+      for (DrillFuncHolder drillFuncHolder : builtInFuncHolderList.get(s)) {
+        String registeredNames[] = drillFuncHolder.getRegisteredNames();
+        String signature = drillFuncHolder.getInputParameters();
+        String returnType = drillFuncHolder.getReturnType().getMinorType().toString();
+        for (String name : registeredNames) {
+          FunctionInfo fnInfo = new FunctionInfo(name, signature, returnType, LocalFunctionRegistry.BUILT_IN);
+          functionMap.put(generateKey(name, signature), fnInfo);
+        }
+      }
+    }
+  }
+
+  //Loads remote-registry functions into functionMap (updates if
+  private void listRemoteFunctionRegistry(Map<String, FunctionInfo> functionMap) {

```

Comment:
@kkhatua 
To avoid this unpleasant parsing I would suggest you add method to `FunctionRegistryHolder` which would return `ListMultimap<String, DrillFuncHolder>`, where key would be jar_name, value would be list of function holders. Let's call it `getAllFunctionsByJars` (name can be changed).   Then in `LocalFunctionRegistry` add delegating method:
```
 public ListMultimap<String, DrillFuncHolder> getAllFunctionsByJars() {
   return registryHolder.getAllFunctionsByJars();
 }
```
Then in `FunctionImplementationRegistry` add the following method:
```
public ListMultimap<String, DrillFuncHolder> getAllFunctionsByJars() {
   if (useDynamicUdfs) {
     syncWithRemoteRegistry(localFunctionRegistry.getVersion());
   }
   return localFunctionRegistry.getAllFunctionsByJars();
 }
```
The above you would get you full list of built-in and remote functions
Unit test can be added to `FunctionRegistryHolderTest` for the new method.

Reply:
Thanks, @arina-ielchiieva . Modified the PR as per your suggestion.